### PR TITLE
Add support for local variable matching during correlation

### PR DESCRIPTION
**Pull request by ChrisFlanTheMan**
and fix issue where multi instance receive tasks were always returning multiple results

### CODE REVIEW

File: messaging-kafka/src/main/java/com/ultimatesoftware/workflow/messaging/kafka/CorrelatingMessageListener.java [hunk 1]
```diff
@@ -100,9 +101,10 @@ private void logResults(String tenantId, String messageType, List<MessageCorrela
                     definitionId = result.getProcessInstance().getProcessDefinitionId();
                     businessKey = result.getProcessInstance().getBusinessKey();
                 } else {
-                    identifier = result.getExecution().getProcessInstanceId();
-                    definitionId = "unknown";
-                    businessKey = "unknown";
+                    ExecutionEntity executionEntity = (ExecutionEntity) result.getExecution();
+                    identifier = executionEntity.getProcessInstanceId();
```
Comment (LeoZ100) [inline messaging-kafka/src/main/java/com/ultimatesoftware/workflow/messaging/kafka/CorrelatingMessageListener.java:105]: Nice, wonder why was this being marked as "unknown" before. Casting to an ExecutionEntity would always work?

File: messaging/src/main/java/com/ultimatesoftware/workflow/messaging/correlation/GenericMessageCorrelator.java [hunk 2]
```diff
@@ -57,12 +60,13 @@ private MessageCorrelationResult executeStartMessageEventCorrelation(Correlation
...
             .forEach(messageCorrelationBuilder::setVariable);
 
-        return correlate(messageCorrelationBuilder.startMessageOnly());
+        return correlate(messageCorrelationBuilder.startMessageOnly(), correlationData);
     }
 
     private List<MessageCorrelationResult> executeCatchMessageEventCorrelation(CorrelationData correlationData) {
         return determineCorrelatableProcessesInstances(correlationData)
             .map((execution -> executeCatchMessageEventCorrelationByExecution(correlationData, execution)))
+            .filter(Objects::nonNull)
```
Comment (LeoZ100) [inline messaging/src/main/java/com/ultimatesoftware/workflow/messaging/correlation/GenericMessageCorrelator.java:69]: Would this still allow for consumers to use "null" values for matching/correlation?

**Review by LeoZ100 — APPROVED** [review]

File: messaging-kafka/src/main/java/com/ultimatesoftware/workflow/messaging/kafka/CorrelatingMessageListener.java (hunk 1 above)
Comment (ChrisFlanTheMan) [reply messaging-kafka/src/main/java/com/ultimatesoftware/workflow/messaging/kafka/CorrelatingMessageListener.java:105]: yeah, I think when Carlos did it before, he didn't realize that the only other option was an execution, which can always be cast to an ExecutionEntity.

File: messaging/src/main/java/com/ultimatesoftware/workflow/messaging/correlation/GenericMessageCorrelator.java (hunk 2 above)
Comment (ChrisFlanTheMan) [reply messaging/src/main/java/com/ultimatesoftware/workflow/messaging/correlation/GenericMessageCorrelator.java:69]: Yep, this only filters out null results, aka if a correlation attempt comes back null

**Pull request merged**